code/modules/reagents/reagent_containers/spray.dm

```diff
@@ -58,9 +59,20 @@
 	if(reagents.has_reagent("lube"))
 		message_admins("[ADMIN_LOOKUPFLW(user)] fired Space lube from \a [src] at [area] [ADMIN_COORDJMP(T)].")
 		log_game("[key_name(user)] fired Space lube from \a [src] at [area] [COORD(T)].")
+	if(reagents.has_reagent("clf3"))
+		message_admins("[ADMIN_LOOKUPFLW(user)] fired CLF3 from \a [src] at [area] [ADMIN_COORDJMP(T)].")
+		log_game("[key_name(user)] fired CLF3 from \a [src] at [area] [COORD(T)].")
+	if(reagents.has_reagent("phlogiston"))
+		message_admins("[ADMIN_LOOKUPFLW(user)] fired phlogiston from \a [src] at [area] [ADMIN_COORDJMP(T)].")
+		log_game("[key_name(user)] fired phlogiston from \a [src] at [area] [COORD(T)].")
+	if(reagents.has_reagent("plasma"))
+		message_admins("[ADMIN_LOOKUPFLW(user)] fired plasma from \a [src] at [area] [ADMIN_COORDJMP(T)].")
+		log_game("[key_name(user)] fired plasma from \a [src] at [area] [COORD(T)].")
+	if(reagents.has_reagent("blackpowder"))

```

Comment:
This could probably just be done with a list of "alarming reagents", and messaging admins if any of the reagents are found in the spraybottle.